okio/src/commonMain/kotlin/okio/internal/FileSystem.kt

```diff
@@ -69,6 +71,23 @@ internal fun FileSystem.commonCopy(source: Path, target: Path) {
   }
 }
 
+fun <T : Any> FileSystem.commonExtend(extensionType: KClass<T>, extension: T): FileSystem {
+  // If this file system is already an extension wrapper, replace it rather than wrapping again.
+  // Note that this optimization doesn't apply to ForwardingFileSystem subclasses, only to the
+  // ForwardingFileSystem base class.
+  if (this::class === ForwardingFileSystem::class) {
```

Comment:
I don't understand why we don't want to, or cannot support subclasses with `this is ForwardingFileSystem` ?

Reply:
This coalesces two ForwardingFileSystem wrappers into one. But it only works if there’s no other methods overridden in the ForwardingFileSystem. If there are, then we shouldn’t replace the users’ subtype with the base type.